### PR TITLE
Pre and Post Script

### PR DESCRIPTION
**Pull request by jessequinn**
modified code to permit the usage of pre and post command scripting.

### CODE REVIEW

File: drone.sh [hunk 1]
```diff
@@ -119,13 +124,21 @@ do
...
+        eval "ssh -p $PORT $USER@$HOST '$prescript'"
+        result=$(($result+$?))
+        echo $(printf "%s" "$ ssh -p $PORT $USER@$HOST result: $?")
+        if [ "$result" -gt "0" ]; then exit $result; fi
+    fi
     eval "$(printf "$expr" "$PORT") $USER@$HOST:$PLUGIN_TARGET"
     result=$(($result+$?))
     if [ "$result" -gt "0" ]; then exit $result; fi
-    if [ -n "$PLUGIN_SCRIPT" ]; then
+    if [ -n "$PLUGIN_POSTSCRIPT" ]; then
```
Comment (mjwwit) [inline drone.sh]: The removal of the `script` option will break every existing usage of this plugin that used that option. I'd love to see a way to keep that functionality intact.

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
-FROM alpine:3.4
-MAINTAINER Michael de Wit <michael@drillster.com>
+FROM alpine:latest
```
Comment (mjwwit) [inline Dockerfile]: Using a `latest` version like this is very dangerous, you may get a version that doesn't work the way you want it to. I'd prefer to see a fixed version. Also, is changing the Alpine version required for this PR?

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+MAINTAINER Jesse Quinn <me@jessequinn.info>
```
Comment (mjwwit) [inline Dockerfile]: I love the work you've done for this project, but does that suddenly make you the maintainer? 😅

File: Dockerfile [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
 
 RUN apk add --no-cache ca-certificates bash openssh-client rsync
-COPY upload.sh /usr/local/
+COPY drone.sh /usr/local/
```
Comment (mjwwit) [inline Dockerfile]: Why did you rename this file? I feel like `upload.sh` was a better match for what the script actually does.

File: drone.sh [hunk 5]
```diff
@@ -100,10 +100,15 @@ fi
...
-# Parse SSH commands
+# Parse SSH precommands
 function join_with { local d=$1; shift; echo -n "$1"; shift; printf "%s" "${@/#/$d}"; }
-IFS=','; read -ra COMMANDS <<< "$PLUGIN_SCRIPT"
-script=$(join_with ' && ' "${COMMANDS[@]}")
+IFS=','; read -ra COMMANDS <<< "$PLUGIN_PRESCRIPT"
+prescript=$(join_with ' && ' "${COMMANDS[@]}")
+
+# Parse SSH postcommands
+function join_with { local d=$1; shift; echo -n "$1"; shift; printf "%s" "${@/#/$d}"; }
```
Comment (mjwwit) [inline drone.sh]: You don't have to define this function twice if you want to use it twice...

File: Dockerfile (hunk 2 above)
Comment (jessequinn) [reply Dockerfile]: this was changed for myself as i am planning to rewrite the plugin to my specific needs just didn’t realize the PR would use my last commit there. same goes for the maintainer thing. i’ll update the code to your request.

File: drone.sh (hunk 1 above)
Comment (jessequinn) [reply drone.sh]: returned to PLUGIN_SCRIPT to avoid breaking

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+/.idea/
```
[the remark this answers is not in the document]
Comment (jessequinn) [reply .gitignore]: removed as per your request

File: Dockerfile (hunk 4 above)
Comment (jessequinn) [reply Dockerfile]: renamed as per your request.

File: drone.sh (hunk 5 above)
Comment (jessequinn) [reply drone.sh]: removed duplicate function.

File: Dockerfile (hunk 3 above)
Comment (jessequinn) [reply Dockerfile]: returned to original name.
